route/people-route.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const Router = require('express').Router;
+const jsonParser = require('body-parser').json();
+const createError = require('http-errors');
+const People = require('../model/people.js');
+const peopleRouter = module.exports = new Router();
+const debug = require('debug')('people:people-route');
+
+peopleRouter.post('/api/people', jsonParser, function(req, res, next) {
+  req.body.timestamp = new Date();
+  new People(req.body).save()
+  .then(people => res.json(people))
+  .catch( err => next(err));
+});
+
+peopleRouter.get('/api/people/:id', function(req, res, next) {
+  People.findById(req.params.id)
+  .populate('president')
+  .then(people => res.json(people))
+  .catch( err => next(createError(404, err.message)));
+});
+
+peopleRouter.get('/api/people/', function(req, res, next) {
+  People.findById()
+  .then(people => res.json(people))
+  .catch( err => next(err));
+});
+
+peopleRouter.put('/api/people/:id', jsonParser, function(req, res, next) {
+  People.findByIdAndUpdate(req.params.id, req.body, {new: true})
+    .then(people => res.json(people))
+    .catch( err => {
+      if(err.name === 'ValidationError') return next(err);
```

Comment:
same issue as above. You aren't properly catching the error returned from mongo.
If you want to catch multiple errors you can just make sure you are catching them correctly and passing the right error status and message.

Example:
``` javascript
.catch( err => {
    if(err.name === 'ValidationError') return next(createError(400, err.message));
    if(err.name === 'CastError') return next(createError(404, err.message));
    next(err);
  });
```